src/lib/ui.js

```diff
@@ -0,0 +1,40 @@
+import config from '../config'
+import {awaitMessage} from '../utils/async'
+
+const uiMethods = async({ method, href } = {}) => {
+  switch (method) {
+    case 'share':
+      if (typeof href === 'string') {
+        // we didn't do any validation but left it for sharing page to validate it.
+        const dialog = openWindow(encodeURIComponent(href))
+        return awaitMessage(dialog)
+      } else {
+        throw Error(`href: ${href} is not valid.`)
+      }
+      break
+    default:
+      throw Error(`method: ${method} is not valid`)
+  }
+}
+
+function openWindow (href) {
+  const w = 500
+  const h = 500
+  const pathForShareToMigme = '/share_to_migme?referrer=&campaign=&return_url=&href='
+  const url = `${config.host}${pathForShareToMigme}${href}`
```

Comment:
Could url-template for doing this since it is part of the package.
```js
const shareUrl = `{host}/share_to_migme{?${['referrer', 'campaign', 'return_url', 'href']}}`
urltemplate.parse(shareUrl).expand({
  href,
  host: config.host,
})
```